.gitlab/benchmarks.yml

```diff
@@ -9,13 +10,40 @@ include:
 
 variables:
   MACROBENCHMARKS_CI_IMAGE: 486234852809.dkr.ecr.us-east-1.amazonaws.com/ci/benchmarking-platform:cpp-nginx
+  BUILD_IMAGE: 486234852809.dkr.ecr.us-east-1.amazonaws.com/images/mirror/b1o7r7e0/nginx_musl_toolchain
+  NGINX_VERSION: 1.26.0
+
+build-nginx-module:
+  stage: build
+  tags: ["arch:amd64"]
+  timeout: 20min
+  image: $BUILD_IMAGE
+  variables:
+    WAF: OFF
+    COVERAGE: OFF
+    RUM: OFF
+    BUILD_TYPE: Release
+    ARCH: x86_64
+  script:
+    - git submodule sync && git submodule update --init --recursive
+    - export NGINX_SRC_DIR="$PWD/nginx"
+    - make build-musl-aux
+  artifacts:
+    name: "artifacts"
+    when: always
+    paths:
+      - .musl-build/ngx_http_datadog_module.so
+    expire_in: 3 months
+
 
 .benchmarks:
   stage: run-benchmarks
+  needs: ["build-nginx-module"]
   tags: ["runner:apm-k8s-same-cpu"]
   timeout: 1h
   image: $MACROBENCHMARKS_CI_IMAGE
   script:
+    - cp .musl-build/ngx_http_datadog_module.so /usr/lib/nginx/modules/ngx_http_datadog_module.so
```

Comment:
I understand that this stage now triggers a build and uses that artifact to be tested, but what I don't understand is where the change is made that will trigger a GitLab pipeline to run on every commit instead of every release, which I don't even know if it's automatically done or manually done. Can you help me understand that?

Reply:
Ah I missed the fact that this is included in `~/.gitlab-ci.yml` and triggered when `$CI_COMMIT_REF_NAME == "master"`. Makes sense now 👍🏼